src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -224,8 +224,10 @@ void WorkStealingThreadPool::PostFork() { pool_->Postfork(); }
 // -------- WorkStealingThreadPool::WorkStealingThreadPoolImpl --------
 
 WorkStealingThreadPool::WorkStealingThreadPoolImpl::WorkStealingThreadPoolImpl(
-    size_t reserve_threads)
-    : reserve_threads_(reserve_threads), queue_(this) {}
+    size_t reserve_threads, size_t max_thread_count)
+    : reserve_threads_(reserve_threads),
```

Comment:
Shouldn't the number of reserve threads be limited to the max thread count?
```suggestion
    : reserve_threads_(std::min(reserve_threads, max_thread_count)),
```